### PR TITLE
Added `Has()` method to Yoda.

### PR DESCRIPTION
**Pull request by mzhurovich**
@dkorolev PTAL.

### CODE REVIEW

File: Sherlock/yoda/container/matrix/api.h [hunk 1]
```diff
@@ -101,6 +101,10 @@ class InnerMapAccessor final {
     }
   }
 
+  bool Has(bricks::copy_free<INNER_KEY> subkey) const {
```
Comment (dkorolev) [inline Sherlock/yoda/container/matrix/api.h]: `s/copy_free/CF` ?

File: Sherlock/yoda/metaprogramming.h [hunk 2]
```diff
@@ -163,11 +163,18 @@ struct YodaData {
 
   template <typename K>
   EntryWrapper<typename CWT<YodaContainer<YT>, type_inference::YETFromK<bricks::rmconstref<K>>>::T_ENTRY> Get(
-      K&& key) {
+      K&& key) const {
```
Comment (dkorolev) [inline Sherlock/yoda/metaprogramming.h]: :+1:

File: Sherlock/yoda/metaprogramming.h [hunk 3]
```diff
@@ -163,11 +163,18 @@ struct YodaData {
...
     typedef bricks::rmconstref<K> SAFE_K;
     return Accessor<CWT<YodaContainer<YT>, type_inference::YETFromK<SAFE_K>>>().Get(std::forward<K>(key));
   }
 
+  template <typename K>
+  bool Has(K&& key) const {
+    typedef bricks::rmconstref<K> SAFE_K;
```
Comment (dkorolev) [inline Sherlock/yoda/metaprogramming.h]: `s/SAFE_KEY/DECAYED_KEY/` ?

File: Sherlock/yoda/metaprogramming.h [hunk 4]
```diff
@@ -308,9 +315,21 @@ struct APICalls {
...
+  // It itself makes a copy of the key to query, and passing in a non-decayed type
+  // enables using `std::forward<>`, choosing between copy and move semantics at compile time.
+  template <typename DATA, typename YET, typename UNDECAYED_KEY>
+  struct TopLevelHas {
+    const bricks::rmconstref<UNDECAYED_KEY> key;
+    TopLevelHas(UNDECAYED_KEY&& key) : key(std::forward<UNDECAYED_KEY>(key)) {}
+    bool operator()(DATA data) const { return YET::Accessor(data).Has(key); }  // TODO(dkorolev): Use `std::move()` here.
+  };
+
+
```
Comment (dkorolev) [inline Sherlock/yoda/metaprogramming.h]: Two extra vertical whitespace.

**Comment by mzhurovich** [comment]
@dkorolev Thanks for the review! Take another look, when you have a minute, pls.

**Comment by dkorolev** [comment]
:+1: LGTM, merging!

**Pull request merged**